src/Nethermind/Nethermind.Core2.Json/JsonNamingPolicySnakeCase.cs

```diff
@@ -0,0 +1,57 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Text;
+using System.Text.Json;
+
+namespace Nethermind.Core2.Json
+{
```

Comment:
https://www.stevejgordon.co.uk/creating-strings-with-no-allocation-overhead-using-string-create-csharp

Reply:
Hmm. snake_case is different from camelCase, because it may insert extra characters. This means to calculate the length would need two passes/iterations over the name.

Although I guess maybe two reads is faster than two allocations.

:thumbsup:

Reply:
not sure if it was in a hot path - if this is used for everys ingle json serialization then it may be worth it, otherwise I suggest merging and leaving for later

Reply:
Maybe we could use Utf8Json here as fast as possible saving some trouble for later?

Reply:
@LukaszRozmej most of the classes the namespace are using Utf8; the only one which is not is the bytes to hex converter (which is actually a big part).  JsonNamingPolicy only uses strings, which I guess they internally cache (I haven't checked). The change of bytes to hex converter to use Utf8 directly would be the biggest improvement... but leave for later. (I'll add a work item)

... Besides, the JSON isn't actually being used yet (only for a few debug places), as the nswag tool only supports the old Newtonsoft Json. I have kept it because I will probably need to change nswag to properly support the OAPI interface.

Reply:
ok, we are coming from eth1 where json is used a lot and had significant perf implications for users - probably best not too focus so much on json here then